src/nnue/evaluate_nnue.cpp

```diff
@@ -154,10 +157,15 @@ namespace Stockfish::Eval::NNUE {
     ASSERT_ALIGNED(transformedFeatures, alignment);
     ASSERT_ALIGNED(buffer, alignment);
 
-    featureTransformer->transform(pos, transformedFeatures);
-    const auto output = network->propagate(transformedFeatures, buffer);
+    const std::size_t bucket = (popcount(pos.pieces()) - 1) / 4;
```

Comment:
Nit: `pos.count<ALL_PIECES>()` is probably faster on most machines than `popcount(pos.pieces())`.